p2pnetwork/node.py

```diff
@@ -152,6 +155,10 @@ def connect_with_node(self, host, port):
             the id's of the node. First we send our id and then we receive the id of the node we are connected to.
             When the connection is made the method outbound_node_connected is invoked.
             TODO: think wheter we need an error event to trigger when the connection has failed!"""
+
+        if self.check_max_connections():
+            return False
```

Comment:
Should be also cap outgoing connections? I was thinking to restrict only incomming connections. What do you think?

Reply:
I think the max connections are to prevent the overlond in the node. Think that it is possible that the library user will automate the outbound connections and it may be necessary to block them all. Anyway we can wait the return of community about this feature.

Reply:
My reason not to block this, is because outgoing connections are initiated by the node itself (or user/programmer) and the incoming connections are initiates from other nodes. I would not like to restrict the node to make connections to other nodes.

Reply:
All right I understand. Maybe if the programer want to block outbound connection he can implement it in the application level.

To change this just change line 209

`if len(self.all_nodes) >= self.max_connections & self.max_connections != 0:`

to 

`if len(self.nodes_inbound) >= self.max_connections & self.max_connections != 0:`

Maybe alter the variable name to `max_inbound_connections`?